quinn-proto/src/tests/token.rs

```diff
@@ -23,6 +23,40 @@ fn stateless_retry() {
     assert_eq!(pair.server.known_cids(), 0);
 }
 
+#[test]
+fn stateless_retry_token_expired() {
+    let _guard = subscribe();
+
+    let fake_time = Arc::new(FakeTimeSource::new());
+
+    let mut pair = Pair::default();
+    pair.server.handle_incoming = Box::new(validate_incoming);
+
+    let mut config = server_config();
+    config.time_source(Arc::clone(&fake_time) as _);
+    pair.server.set_server_config(Some(Arc::new(config)));
+
+    let client_ch = pair.begin_connect(client_config());
+    pair.drive_client();
+    pair.drive_server();
+    pair.drive_client();
+
+    // to expire retry token
+    fake_time.advance(Duration::from_secs(100));
```

Comment:
Would it be better to test that it counts as expired even if it's just a tiny bit late, like a second? The retry expiration period is configurable in the config.

Reply:
As you mentioned, I will update this to wait for the retry token lifetime to expire without using `fake_time`.

Reply:
I still think using fake time is a good idea! I just think that 100 seconds is probably too high, and more importantly, arbitrary.

Reply:
I see what you meant in your previous comment!
I've updated the test to verify expiration by advancing fake time to the retry token lifetime plus a small additional duration.
https://github.com/quinn-rs/quinn/pull/2149/files#diff-85a9a9679a7643217b4f2d5e0af2db99104ab928e995e102163ef62cc6b02fc5R48